modules/rdf_export/src/Encoder/RdfEncoder.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Drupal\rdf_export\Encoder;
+
+use EasyRdf\Format;
+use Symfony\Component\Serializer\Encoder\EncoderInterface;
+
+/**
+ * Adds RDF encoder support for the Serialization API.
+ */
+class RdfEncoder implements EncoderInterface {
```

Comment:
I’m concerned by how this could be extended by a 3rd party module that wants to add a new format. Right now he has to swap the service because the allowed format list is hardcoded in `RdfEncoder::$supportedFormats`. But what if we can make it dynamic? When the container is compiled there is a pass that creates a `serializer.formats` service container parameter. That can be easily retrieved with `\Drupal::getContainer()->getParameter('serializer.formats')` and then intersected with EasyRdf list, so we get the list of RDF formats supported by the encoder. Then a 3rd party module only needs to add its service with the new format:

```
my_module.encoder.formatx:
    class: Drupal\rdf_export\Encoder\RdfEncoder
    tags:
      - { name: encoder, format: formatx }
```
and `RdfEncoder::supportedFormats()` will become:

```
public static function supportedFormats(): array {
    $registered_formats = \Drupal::getContainer()->getParameter('serializer.formats');
    $formats = Format::getFormats();
    /** @var \EasyRdf\Format[] $supported_formats */
    return array_intersect($formats, $registered_formats);
}
```